### DISCUSSION TITLE
Scaling along a non-cartesian axis

### DISCUSSION THREAD

**Discussion by tpolakovic** [bevyengine/bevy#7850] [Q&A] [answered]
I have a 2D mesh (for testing purposes, will most probably become a sprite eventually) that I want to "squish" along a vector that's not parallel to the X/Y axis, and changes at every time step.

Outside of Bevy, rotate/scale/rotate back, but I'm not sure how to achieve this here.

**Comment by Anti-Alias** [comment, marked as answer]
I believe this is only possible via parenting.
So, make 2 entities: 1 parent, and 1 child with the 2D mesh.

Spawn the parent, and spawn the child as a "child" of parent.
Then, rotate the parent and scale the child accordingly.
Not super elegant, but possible.
This same restriction applies to Unity, I believe.

See this: https://answers.unity.com/questions/13436/how-to-scale-an-object-along-an-arbitrary-axis-in.html

**Reply by tpolakovic** [reply]
Thanks for the hint. It does seem to work with minor modifications: You scale and rotate the parent and counter-rotate the child. This leaves the object at the same global rotation and scales along the arbitrary axis.

**Reply by alice-i-cecile** [reply]
Modifying the GlobalTransform directly might have the same effect.

**Reply by tpolakovic** [reply]
> Modifying the GlobalTransform directly might have the same effect.

That also works and is arguably more ergonomic.
